### ISSUE TITLE
Allow ShowAndWaitAsync task to be cancellable (Using CancellationToken)

### ISSUE THREAD

**Issue by PerikiyoXD** [lepoco/wpfui#470]
### Is your feature request related to a problem? Please describe

Basically, I'm writing a multitask await scenario.

I'm trying to use a standard Task, and Task<IDialogControl.ButtonPressed>.

The objective is use the popup to block interactions until either, the Task is finished or the popup button is pressed.

The approach I use is wait for either task to be finished and cancel the other.

I can't cancel the ShowAndWaitAsync Task.

### Describe the solution you'd like

Let the ShowAndWaitAsync take a cancellation token to allow cancelling the interaction.

### Describe alternatives you've considered

I tried to do this with events and it's not a viable workaround.

I'm just using a single function to wait until "Task finishes or Click happens".

### Additional context

Example code which adds context for the desired execution.

```cs
CancellationTokenSource _myCancellationTokenSource = new CancellationTokenSource();

Task<IDialogControl.ButtonPressed> popupInteractedTask = _dialogControl.ShowAndWaitAsync("Blocked", "Interact to unblock" /*, _myCancellationTokenSource.Token  */);

Task taskEndedTask = /* arbitrary task which takes _myCancellationTokenSource.Token */;

// Create an array of tasks
Task[] tasks = new Task[] { popupInteractedTask, taskEndedTask };

// Task that waits for any of the tasks on the array to end
Task<int> resultWaitTask = new Task<int>(() =>
{
    return Task.WaitAny(tasks);
});
resultWaitTask.Start();

// Non-blockingly wait until any task ended
await resultWaitTask;

// Do cleanup of the non-finished task

// This works fine for the taskEndedTask task, but not for popupInteractedTask.
_myCancellationTokenSource.Cancel();

```

**Comment by PerikiyoXD** [comment]
I ask that it be reviewed and/or implemented as soon as possible.

Understand that this is not a requirement and I am only punctuating it.

**Comment by PerikiyoXD** [comment]
I'll sponsor if the enhancement is made
